sonar-plugin-api/src/main/java/org/sonar/api/web/page/Page.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.api.web.page;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.CheckForNull;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * @since 6.3
+ */
+public final class Page {
+  private final String key;
+  private final String name;
+  private final boolean isAdmin;
+  private final boolean isGlobal;
+  private final List<Qualifier> qualifiers;
+
+  private Page(Builder builder) {
+    this.key = builder.key;
+    this.name = builder.name;
+    this.qualifiers = Stream.of(builder.qualifiers).sorted().collect(Collectors.toList());
+    this.isAdmin = builder.isAdmin;
+    this.isGlobal = builder.isGlobal;
+  }
+
+  public static Builder builder(String key, String name) {
```

Comment:
`name` should not be a mandatory field. It should be loaded from l10n bundle by default. Moreover keeping a single parameter is better for forward-compatibility.